app/models/quota.rb

```diff
@@ -18,8 +20,39 @@ def find(quota_path, user)
       user  = user && user.to_s
 
       quotas = []
+
+      # Attempt to convert path into a URI
+      if quota_path.instance_of?(String) and quota_path.match(/^https?:/)
+        uri = URI.parse(quota_path)
+        # If it is a URI, and it is http:// or https://
+        begin
+          raw = Net::HTTP.get(uri)
+        rescue Exception => e
+            # There are a million ways this could go wrong, assume configured correctly and is temporary issue (e.g., not a URL typo).
+            Rails.logger.error("Quota URI failed to return data: #{e.message}")
+            # Bail with empty results. Don't break portal because web service is down.
+            return []
+        end
+      else
+        # If not a URL, assume it is a local file and attempt to read.
+        # If we're fed a string, convert to Pathname. Otherwise, use as is.
+        if quota_path.instance_of?(String)
+          quota_path = Pathname.new(quota_path)
+        end
+        # Assume this always works, unless configured wrong, in which case don't attempt to catch.
+        raw = quota_path.read
+      end
+
+      # Attempt to parse raw JSON into an object
+      begin
+        json = JSON.parse(raw)
+      rescue JSON::ParserError => e
+        Rails.logger.error("Quota file is not limited JSON: #{e.message}")
```

Comment:
If reading the json string fails, `json` will be nil, but we will attempt to use it in the case statement below, which will result in an exception.

Simplest fix is to `return []` similar to what you do above when handling exception raised by `Net::HTTP.get(uri)`

Reply:
Good point, will do that.